src/components/timeline/Row.svelte

```diff
@@ -362,72 +399,127 @@
     overlaySvgSelection.call(zoom.transform, timelineZoomTransform);
   }
 
-  $: if (activityLayers && spansMap && activityDirectives && typeof filterActivitiesByTime === 'boolean') {
-    activityTree = [];
-    idToColorMaps = { directives: {}, spans: {} };
-
-    let spansList = Object.values(spansMap);
-    const directivesByType = groupBy(activityDirectives, 'type');
-    const spansByType = groupBy(spansList, 'type');
-    if (activityLayers.length) {
-      let directives: ActivityDirective[] = [];
-      let spans: Span[] = [];
-
-      // track directives and spans that have been seen to avoid double counting
-      // if more than one layer matches a type
-      let seenDirectiveIds: Record<number, boolean> = {};
-      let seenSpanIds: Record<number, boolean> = {};
-      activityLayers.forEach(layer => {
-        if (layer.filter && layer.filter.activity !== undefined) {
-          const types = layer.filter.activity.types || [];
-          types.forEach(type => {
-            const matchingDirectives = directivesByType[type];
-            if (matchingDirectives) {
-              const uniqueDirectives: ActivityDirective[] = [];
-              matchingDirectives.forEach(directive => {
-                if (!seenDirectiveIds[directive.id]) {
-                  idToColorMaps.directives[directive.id] = layer.activityColor;
-                  seenDirectiveIds[directive.id] = true;
-                  uniqueDirectives.push(directive);
-                }
-              });
-              directives = directives.concat(uniqueDirectives);
-            }
-            const matchingSpans = spansByType[type];
-            if (matchingSpans) {
-              const uniqueSpans: Span[] = [];
-              matchingSpans.forEach(span => {
-                if (!seenSpanIds[span.span_id]) {
-                  idToColorMaps.spans[span.span_id] = layer.activityColor;
-                  seenSpanIds[span.span_id] = true;
-                  uniqueSpans.push(span);
-                }
-              });
-              spans = spans.concat(uniqueSpans);
+  $: if (
+    (activityLayers && spansMap && activityDirectives && typeof filterItemsByTime === 'boolean') ||
```

Comment:
FilterItemsByTime doesn't appear to be working for external events.

Reply:
Handled in [fix filterByTime for events](https://github.com/NASA-AMMOS/aerie-ui/commit/9acea3e069630b3cf50db37bccfe26eb4a5a9dd0).